src/plugins/dashboard_embeddable_container/public/plugin.tsx

```diff
@@ -70,6 +70,7 @@ export class DashboardEmbeddableContainerPublicPlugin
     };
 
     const ExitFullScreenButton: React.FC<ExitFullScreenButtonProps> = props => {
+      // TODO: Only call useHideChrome if chrome is visible
       useHideChrome();
```

Comment:
This line is the cause of the main issue remaining with the "show top nav menu" option. When called it hides the chrome (and I've made my changes react to it as well) which is fine as it should already be hidden in embedded mode, but when exiting full screen mode it calls the clean up callback which always displays the chrome. Ideally this would revert back to the previous state, or just skip the whole function if `isVisible$` is false, but I wasn't sure the best way of doing this with RxJS. It feels like a hack if I have to subscribe to get the current value and then unsubscribe immediately afterwards. I'm not very familiar with RxJS, so what would be the correct "reactive" way of doing this?